pkg/pluginterfaces/pluginterfaces.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2022 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pluginterfaces
+
+import (
+	"context"
+	"net/http"
+
+	"go.uber.org/zap"
+)
+
+// Any logger of this interface can be used by rtplugs and all connected plugs
+type Logger interface {
+	Debugf(format string, args ...interface{})
```

Comment:
Do we need to define our own interface for logging? I am guessing (could be wrong) that an existing logging interface exist in serving or Knative pkg?

Reply:
True, Knative uses zap logger. 
But the security plugs are designed to enable using them in more generic Kubernetes use cases where other loggers may be used. The approach taken is to define a logger interface (that is much needed and unfortunately missing in golang)